### ISSUE TITLE
GridSearch

### ISSUE THREAD

**Issue by vatsalag99** [JustGlowing/minisom#28]
Hi, thanks for the library! Is there a way of integrating this code with GridSearch to find the optimal parameters to minimize the quantization error?

**Comment by vatsalag99** [comment]
I also had a general question of what solutions there are to separate overlapping clusters after training the SOM. I am still a little unfamiliar with this type of model, so any suggestions would be great!

**Comment by JustGlowing** [comment]
If you write a wrapper for MiniSom around sklearn Estimators, you'll be able to use MiniSom with GridSearch.

I don't understand your question about the overlapping clusters.

**Comment by vatsalag99** [comment]
Thanks! To clarify my second question, I'm unsure of how to minimize the number of neurons that are the winner for multiple classes. This would be to increase the accuracy of predicting clusters for test data.

**Comment by JustGlowing** [comment]
Hi again, that depends on you data and the parameters used to train the som. I'd run a GridSearch trying to maximize the accuracy.

Please leave a start if you like MiniSom!

**Comment by vatsalag99** [comment]
Thanks for the advice! I haven't created wrappers around estimators before, so are there any tips you could give for how to approach this?

**Comment by JustGlowing** [comment]
hi Vatsal, here's explained how to create estimators: https://scikit-learn.org/stable/developers/contributing.html#rolling-your-own-estimator

**Comment by vatsalag99** [comment]
Thank you so much! I have one last question regarding predictions. How I could get obtain the corresponding label in the training data for the coordinates returned by winner()?

**Comment by JustGlowing** [comment]
Assuming that each neuron is a cluster for you, you just need to unravel the index of the neuron on the map. This function can help you: https://docs.scipy.org/doc/numpy/reference/generated/numpy.unravel_index.html
